REQUIREMENTS.md

```diff
@@ -0,0 +1,6 @@
+# Requirements
+
+You must have the following dependencies installed on your system to work with
+  this repository:
+
+- [Docker](https://docs.docker.com/get-started/get-docker/)
```

Comment:
@vikramnitin9 I'd also like to disambiguate [this line about requiring an `LLM_API_KEY`](https://github.com/vikramnitin9/rust_verification/blob/2285aee48b1cb1845e529b1a5fe464bea24db871/README.md?plain=1#L12) from the README.

Is this a Vertex API Key?